### PR TITLE
Tests: Remove thread leak filter

### PR DESCRIPTION
**Pull request by rjernst**
Now that the global cluster is gone, we can get rid of this filter that allows leaks across tests.  This change uncovered one thread pool that we never closed, as well as leaks across tests for suite scope. 

There is still one leak, when running HunspellServiceTests (it leaks the `timer` and `scheduler` threadpools). Perhaps someone has ideas on what could be happening there.

### CODE REVIEW

**Comment by s1monw** [comment]
awesome LGTM

File: src/main/java/org/elasticsearch/indices/recovery/RecoverySettings.java [hunk 1]
```diff
@@ -145,6 +145,7 @@ public RecoverySettings(Settings settings, NodeSettingsService nodeSettingsServi
     @Override
     public void close() {
         ThreadPool.terminate(concurrentStreamPool, 1, TimeUnit.SECONDS);
+        ThreadPool.terminate(concurrentSmallFileStreamPool, 1, TimeUnit.SECONDS);
     }
```
Comment (s1monw) [inline src/main/java/org/elasticsearch/indices/recovery/RecoverySettings.java]: can we port this to 1.4?

Comment (rjernst) [inline src/main/java/org/elasticsearch/indices/recovery/RecoverySettings.java]: Done.

**Comment by javanna** [comment]
Shouldn't we remove `ElasticsearchThreadFilter` as well now? It seems unused at this point.

**Comment by javanna** [comment]
Also, @rjernst can you please label this PR?

**Comment by dadoonet** [comment]
If we do remove it as per @javanna suggestion, please update also [Twitter River](https://github.com/elasticsearch/elasticsearch-river-twitter/) project in `master`, `es-1.x` and `es-1.4` branches: https://github.com/elasticsearch/elasticsearch-river-twitter/blob/master/src/test/java/org/elasticsearch/river/twitter/test/TwitterIntegrationTest.java#L84

**Comment by dadoonet** [comment]
My comment was unclear. I meant that we just have to remove it from `TwitterIntegrationTest` and replace:

```java
@ThreadLeakFilters(defaultFilters = true, filters = {ElasticsearchThreadFilter.class, Twitter4JThreadFilter.class})
```

by:

```java
@ThreadLeakFilters(defaultFilters = true, filters = {Twitter4JThreadFilter.class})
```

**Comment by rjernst** [comment]
I pushed the removal of ElasticsearchThreadFilter in e221dc2 and 28776e6, and also fixed the twitter river as David noted.

**Comment by dadoonet** [comment]
Interesting. I was not aware of this: https://github.com/elasticsearch/elasticsearch-analysis-icu/blob/master/src/test/java/org/elasticsearch/index/analysis/TestIndexableBinaryStringTools.java#L43

Thank you Jenkins catching it! :) I'm going to fix

**Pull request closed without merging**